Topic1/src/Client.java

```diff
@@ -0,0 +1,87 @@
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+* @author Melina
+* @version 1.0 
+*/ 
+
+public class Client {
+	private String name;
+	private int creditNumber;
+	private String email;
+	private String pass;
+
+	private List<ShoppingCart> carts;
+	
+	public Client(String name, int creditNumber, String email, String pass) {
+		this.name=name;
+		this.creditNumber=creditNumber;
+		this.email=email;
+		this.pass=pass;
+		carts=new ArrayList<ShoppingCart>();
+	}
+	
+	//make the payment transaction
+	public void payment(int c) {
+		System.out.println("Payment transaction ");
+		carts.get(c).payCart();;
+	}
+	
+	public String showCart(String idCart) {
+		for (ShoppingCart cart: carts){
+			if (idCart.equals(cart.getIdCart()))
+				return cart.showProducts();
+		}
+		return null;
+	}
+
+	public void addToCart(int c,IOffer p) {
```

Comment:
Warning: The addToCart method is never called, but assuming it was... then the caller would have to know in which position to store the Offer. I wonder how the caller would know this...